Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp

```diff
@@ -156,8 +156,8 @@ void WebXROpaqueFramebuffer::startFrame(const PlatformXR::Device::FrameData::Lay
     auto colorTextureSource = makeEGLImageSource({ data.surface->createSendRight(), false });
     auto depthStencilBufferSource = makeEGLImageSource({ MachSendRight(), false });
 #elif USE(MTLTEXTURE_FOR_XR_LAYER_DATA)
-    auto colorTextureSource = makeEGLImageSource(data.colorTexture);
-    auto depthStencilBufferSource = makeEGLImageSource(data.depthStencilBuffer);
+    auto colorTextureSource = makeEGLImageSource({ WTFMove(data.colorTexture.handle), data.colorTexture.isShared });
```

Comment:
Would it be more appropriate to still try to live with the existing `std::pair`, and rather solve by having `startFrame` accept `LayerData&& data`?